common/config/elasticsearch.go

```diff
@@ -55,6 +55,8 @@ type (
 		AWSSigning AWSSigning `yaml:"awsSigning"`
 		// optional to use Signed Certificates over https
 		TLS TLS `yaml:"tls"`
+		// optional to add custom headers
+		CustomHeaders CustomHeaders `yaml:"customHeaders"`
```

Comment:
I don't have a huge amount of context about this, but generally headers are modeled as untyped because they're dealing with random data not forseen by the original authors. 

Me being lazy, I'd typically pick a `map[string]string` to solve that, though as @Groxx  reminds me occasionally, this is not a strictly accurate representation of headers.

Reply:
Sure, I can make it map[string]string, I have done the same thing for kafka properties. 

The context for this header is we need to pass custom headers when we use OS client in monorepo, OSS is not requiring that

Reply:
Updated!